tardis/io/tests/test_config_reader.py

```diff
@@ -367,5 +368,36 @@ def test_absolute_relative_config_paths(monkeypatch, cwd):
     config_abs = config_reader.Configuration.from_yaml(os.path.abspath(data_path(filename)),
                                                        test_parser=True)
 
+def test_custom_yaml_loader():
+    filename = 'tardis_configv1_density_exponential_test.yml'
+    default_conf = config_reader.Configuration.from_yaml(data_path(filename), test_parser=True,
+                                                         loader=yaml.Loader)
+    custom_conf = config_reader.Configuration.from_yaml(data_path(filename), test_parser=True,
+                                                        loader=config_reader.YAMLLoader)
+    assert len(default_conf) == len(custom_conf)
+    assert set(default_conf.keys()) == set(custom_conf.keys())
+
+    def check_equality(item1, item2):
```

Comment:
I think defining this function in a global namespace would make more sense.

Additionally I think splitting this function into two logical parts would be useful. You would have one function that's responsible for traversing the dictionary and another function which does the comparing. That way you separate the two different parts of the function.
For the traversing part it should be save to use something similiar to [http://stackoverflow.com/questions/3860813/recursively-traverse-multidimensional-dictionary-dimension-unknown](url). You just need to modify that a bit to allow traversing of two dictionaries at once.